certbot/certbot/_internal/main.py

```diff
@@ -800,7 +800,13 @@ def _install_cert(config, le_client, domains, lineage=None):
     path_provider = lineage if lineage else config
     assert path_provider.cert_path is not None
 
-    le_client.deploy_certificate(domains, path_provider.key_path,
+    cert_name = None
```

Comment:
Oh boy. So. How did we get to this point in the execution?
1. We came from run and have a perfectly good RenewableCert that gave us a certname.
2. We came from install and the user specified the certname
3. We came from install and the user specified the paths, but they're our normal default paths and the certname will be right there in the file path
4. We came from install but the user just wants to use our installer functionality separate from everything else we do and just gave us paths to files located who-knows-where.

We currently cover cases 1 and 2. We could easily cover case 3 using `cert_manager.cert_path_to_lineage()`, and if we get an error we're probably just in case 4 and can continue as planned.

Reply:
I am puzzled as to how `cert_manager.cert_path_to_lineage(config)` works. It appears to look at `cli_config.cert_path[0]`, which evaluates to `/` if `cert_path` is a `str`. Predictably this results in an error being raised if I pass in `config`:

    No match found for cert-path /

The tests for `cert_path_to_lineage` seem to indicate that `cert_path` is some kind of tuple, but I can't connect the dots in the actual code.

I am hoping that based on your comment [here](https://github.com/certbot/certbot/pull/4914#discussion_r137903643) that you know what is going on :laughing: .

Reply:
Oy. So, that's true when looking at the result of `storage.cert_path_for_cert_name`, which returns a tuple of `(path, contents_at_path)`, which we save into `config.cert_path` [here](https://github.com/certbot/certbot/blob/7399807ff2b4f8ac1f97e4f037f374610ca88797/certbot/certbot/_internal/main.py#L1093), if `cert_path` hasn't been set on the command line. So looks like that's a revoke-specific choice that we make to overload that variable, and has been since at least [February 2016](https://github.com/certbot/certbot/blame/7399807ff2b4f8ac1f97e4f037f374610ca88797/certbot/certbot/_internal/main.py#L1109), which is when I got bored of tracing it. So we could probably just call that here in the same way, or just create a tuple and pass that in.